compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1159,6 +1182,10 @@ impl<'tcx> InferCtxt<'tcx> {
                 Ty::new_var(self.tcx, ty_var_id).into()
             }
             GenericParamDefKind::Const { .. } => {
+                // todo what about using effect var here
+                if self.tcx.has_attr(param.def_id, sym::rustc_host) {
```

Comment:
hmm 🤔 I guess we could add a `is_host_effect: bool` field to `GenericParamDefKind::Const` to make this simpler. it's not going to increase the size of `GenericParamDefKind`, even if we change it to an enum at some point to encode different effect kinds